staging/src/k8s.io/csi-translation-lib/plugins/aws_ebs.go

```diff
@@ -57,6 +58,10 @@ func (t *awsElasticBlockStoreCSITranslator) TranslateInTreeInlineVolumeToCSI(vol
 	}
 	ebsSource := volume.AWSElasticBlockStore
 	pv := &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			// A.K.A InnerVolumeSpecName required to match for Unmount
+			Name: volume.Name,
```

Comment:
Where does this get used?  Since inner volume name for inline is user-specified, what happens if 2 users have the give the same volume name in their Pod spec, but the underlying disks are different?

Reply:
We can see in `Mount` we use the `spec.Name()`:
https://github.com/kubernetes/kubernetes/blob/d4d8daea7368e5fc9ad8c375efff74a65d0f91bf/pkg/volume/csi/csi_plugin.go#L410
to determine the path in which we mount the volume and save the volData. This corresponds to the `PV.name` of the migrated volume.

However, in `Unmount`, we use the `volumeToUnmount.InnerVolumeSpecName` (I assume because in some cases `PV` or `Volume` may have been lost and this info is cached somewhere):
https://github.com/kubernetes/kubernetes/blob/d4d8daea7368e5fc9ad8c375efff74a65d0f91bf/pkg/volume/util/operationexecutor/operation_generator.go#L838
which corresponds to the original `volume.Name` of the inline volume.

And as we can see here, both `PV.name` and `Volume.Name` values are the `InnerVolumeSpecName`:
https://github.com/kubernetes/kubernetes/blob/d4d8daea7368e5fc9ad8c375efff74a65d0f91bf/pkg/volume/util/operationexecutor/operation_executor.go#L455
and therefore the assumption has been made in the code that they are equivalent and that's why I have to make them equivalent here

You cannot give the same volume name for 2 volumes in the same pod as those are the reference to the underlying PD.

Reply:
I mean the same volume name in two different pods.  Is podUID always included in the directory name, dsw, asw, etc?

Reply:
yes, I saw the directory contains pod UID, ASW definitely distinguishes between pods, and I believe dsw does.

Migrated volumes are all treated like their original volumes (in this case the in-tree inline volume) in the internal caches. They are only shimmed for the last mile where CSI needs to understand what to do.